lib/matplotlib/tests/test_backend_qt.py

```diff
@@ -64,50 +54,143 @@ def test_fig_close():
+            buf += c
+            if buf.endswith(terminator):
+                return
+
+
+def _test_sigint_impl(backend, target_name, kwargs):
+    import sys
+    import matplotlib.pyplot as plt
+    import os
+    import threading
+
+    plt.switch_backend(backend)
+    from matplotlib.backends.qt_compat import QtCore
+
+    def interupter():
+        if sys.platform == 'win32':
+            import win32api
+            win32api.GenerateConsoleCtrlEvent(0, 0)
+        else:
+            import signal
+            os.kill(os.getpid(), signal.SIGINT)
+
+    target = getattr(plt, target_name)
+    timer = threading.Timer(1, interupter)
+    fig = plt.figure()
+    fig.canvas.mpl_connect(
+        'draw_event',
+        lambda *args: print('DRAW', flush=True)
+    )
+    fig.canvas.mpl_connect(
+        'draw_event',
+        lambda *args: timer.start()
+    )
+    try:
         target(**kwargs)
+    except KeyboardInterrupt:
+        print('SUCCESS', flush=True)
 
 
 @pytest.mark.backend('QtAgg', skip_on_importerror=True)
 @pytest.mark.parametrize("target, kwargs", [
-    (plt.show, {"block": True}),
-    (plt.pause, {"interval": 10})
+    ('show', {'block': True}),
+    ('pause', {'interval': 10})
 ])
-def test_other_signal_before_sigint(qt_core, platform_simulate_ctrl_c,
-                                    target, kwargs):
-    plt.figure()
+def test_sigint(target, kwargs):
+    backend = plt.get_backend()
+    proc = WaitForStringPopen(
+        [sys.executable, "-c",
+         inspect.getsource(_test_sigint_impl) +
+         f"\n_test_sigint_impl({backend!r}, {target!r}, {kwargs!r})"])
+    try:
+        proc.wait_for('DRAW')
+        stdout, _ = proc.communicate(timeout=_test_timeout)
+    except:
+        proc.kill()
+        stdout, _ = proc.communicate()
+        raise
+    print(stdout)
+    assert 'SUCCESS' in stdout
+
+
+def _test_other_signal_before_sigint_impl(backend, target_name, kwargs):
+    import signal
+    import sys
+    import matplotlib.pyplot as plt
+    plt.switch_backend(backend)
+    from matplotlib.backends.qt_compat import QtCore
 
-    sigcld_caught = False
-    def custom_sigpipe_handler(signum, frame):
-        nonlocal sigcld_caught
-        sigcld_caught = True
-    signal.signal(signal.SIGCHLD, custom_sigpipe_handler)
+    target = getattr(plt, target_name)
 
-    def fire_other_signal():
-        os.kill(os.getpid(), signal.SIGCHLD)
+    fig = plt.figure()
+    fig.canvas.mpl_connect('draw_event',
+                           lambda *args: print('DRAW', flush=True))
 
-    def fire_sigint():
-        platform_simulate_ctrl_c()
+    timer = fig.canvas.new_timer(interval=1)
+    timer.single_shot = True
+    timer.add_callback(print, 'SIGUSR1', flush=True)
 
-    qt_core.QTimer.singleShot(50, fire_other_signal)
-    qt_core.QTimer.singleShot(100, fire_sigint)
+    def custom_signal_handler(signum, frame):
+        timer.start()
+    signal.signal(signal.SIGUSR1, custom_signal_handler)
```

Comment:
There is a whole bunch of code here that is not being run.  Is that on purpose/it will be run at a later date?

Reply:
Everything in `*_impl` functions are run in a subprocess which coverage doesn't see.